src/core/border_router/routing_manager.cpp

```diff
@@ -683,6 +683,53 @@ bool RoutingManager::IsValidOnLinkPrefix(const Ip6::Prefix &aOnLinkPrefix)
            !aOnLinkPrefix.IsMulticast();
 }
 
+bool RoutingManager::IsExternalRoutePublished() const
```

Comment:
A quick note on terminology we use within OT for consistency:

- We use **add/remove** entries to/from network data which does it directly with no additional checks.
- We use **publish/unpublish** entries via the ["Network Data Publisher"](https://github.com/openthread/openthread/blob/main/include/openthread/netdata_publisher.h), which limits the number of similar entries across all BRs.


So it's possible to publish an entry (`NetworkData::Publisher::PublishExternalRoute()`) without it being added by this device in network data if enough other BRs have already added the same prefix.

The method name `IsExternalRoutePublished` does not reflect that it actually checks whether the entry has been added to network data and registered with leader, not whether it is being published.